pkgs/applications/terminal-emulators/wezterm/default.nix

```diff
@@ -0,0 +1,87 @@
+{ rustPlatform
+, lib
+, fetchFromGitHub
+
+, pkgconfig
+, fontconfig
+, python3
+, openssl
+, perl
+
+, dbus
+, libX11
+, xcbutil
+, libxcb
+, xcbutilkeysyms
+, xcbutilwm # contains xcb-ewmh among others
+, libxkbcommon
+, libglvnd # libEGL.so.1
+, egl-wayland
+, wayland
+, libGLU
+, libGL
+, freetype
+, zlib
+}:
+let
+  runtimeDeps = [
+    libX11
+    xcbutil
+    libxcb
+    xcbutilkeysyms
+    xcbutilwm
+    libxkbcommon
+    dbus
+    libglvnd
+    zlib
+    egl-wayland
+    wayland
+    libGLU
+    libGL
+    fontconfig
+    freetype
+  ];
+
+  pname = "wezterm";
+  version = "3bd8d8c84591f4d015ff9a47ddb478e55c231fda";
+in
+rustPlatform.buildRustPackage {
+  inherit pname;
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "wez";
+    repo = pname;
+    rev = "${version}";
+    fetchSubmodules = true;
+    sha256 = "13xf3685kir4p159hsxrqkj9p2lwgfp0n13h9zadslrd44l8b8j8";
+  };
+  cargoSha256 = "0ri79aiipy93dabiwp1jgsy1pgwrdr38n6wrp5j568m7cqbdrmwn";
+
+  nativeBuildInputs = [
+    pkgconfig
+    python3
+    openssl.dev
+    perl
+  ];
+
+  buildInputs = runtimeDeps;
+
+  installPhase = ''
+    for artifact in wezterm wezterm-gui wezterm-mux-server strip-ansi-escapes; do
```

Comment:
I don't know how/if all these translate to nixos, but if you look at the rpm spec generation here:

https://github.com/wez/wezterm/blob/master/ci/deploy.sh#L84-L106

there are a couple of additional artifacts to deploy for linux:

* The `.desktop` file and icon
* The `/etc/profile.d` shell integration scripts

Reply:
Thanks @wez. I added the install instructions although I'm not sure how to verify any of them. Maybe someone else jumps in or we can fix it up subsequently.